### PR TITLE
EDU-880 Create waveform plugin

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-880

### CODE REVIEW

File: src/plugins/audio-waveform/audio-waveform-editor.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                url={storageLocationPathToUrl(sourceUrl)}
+                onUrlChange={handleInternalUrlChange}
+                />
+            </div>
+          </FormItem>
+        )}
+        <Divider plain>{t('generateImageDividerText')}</Divider>
+        <FormItem {...formItemLayoutWithoutLabel}>
+          <Button type="primary" onClick={handleCreateWaveform}>
+            {t('Wellenform aus Audio-Datei erstellen')}
```
Comment (adrianaluchian) [inline src/plugins/audio-waveform/audio-waveform-editor.js:105]: Wrong text here 😅

File: src/plugins/audio-waveform/audio-waveform-info.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import joi from 'joi';
+import React from 'react';
+import cloneDeep from '../../utils/clone-deep.js';
+import AudioDisplay from './audio-waveform-display.js';
```
Comment (adrianaluchian) [inline src/plugins/audio-waveform/audio-waveform-info.js:4]: `AudioWaveformDisplay`

File: src/plugins/audio-waveform/audio-waveform-generator-dialog.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+            <span>{t('dialogLocalFilePickerPenColorLabel')}: </span>
+            <ColorPicker color={waveformPenColor} onChange={setWaveformPenColor} width={382} inline />
+          </div>
+          <div>
+            <span>{t('dialogLocalFilePickerBaselineColorLabel')}: </span>
+            <ColorPicker color={waveformBaselineColor} onChange={setWaveformBaselineColor} width={382} inline />
+          </div>
+          <div>
+            <span>{t('dialogLocalFilePickerBackgroundColorLabel')}: </span>
+            <ColorPicker color={waveformBackgroundColor} onChange={setWaveformBackgroundColor} width={382} inline />
```
Comment (adrianaluchian) [inline src/plugins/audio-waveform/audio-waveform-generator-dialog.js:186]: A very particular number, this 382 😄 What does it mean?

**Comment by adrianaluchian** [comment]
Works very nicely! 👏 

Just 2 tiny things I found while testing:
- the info icon overlaps a bit with the background in the dialog
Screen Shot 2022-10-19 at 09 47 46

- when you select a wrong/non-audio format in the dialog from the CDN, it throws and does not stop the spinner; maybe we should display a message to the user to draw attention to having not picked an audio file and stop the loading spinner

File: src/plugins/audio-waveform/audio-waveform-editor.js (hunk 1 above)
Comment (ahelmberger) [reply src/plugins/audio-waveform/audio-waveform-editor.js]: Oh my! I forgot to add the translation. Fixed! Thanks for spotting it.

File: src/plugins/audio-waveform/audio-waveform-info.js (hunk 2 above)
Comment (ahelmberger) [reply src/plugins/audio-waveform/audio-waveform-info.js:4]: Fixed.

File: src/plugins/audio-waveform/audio-waveform-generator-dialog.js (hunk 3 above)
Comment (ahelmberger) [reply src/plugins/audio-waveform/audio-waveform-generator-dialog.js]: It overrode the default of 320 in all cases :rofl: I changed it so the default is now 382 and the props are not set anywhere. (It's the width of the color swatches dropdown btw).

Comment (ahelmberger) [reply src/plugins/audio-waveform/audio-waveform-generator-dialog.js]: > Works very nicely! clap
> 
> Just 2 tiny things I found while testing:
> 
> * the info icon overlaps a bit with the background in the dialog

I removed this tooltip/icon completely now as it was redundant (the same text is shown in the preview area initially)

> * when you select a wrong/non-audio format in the dialog from the CDN, it throws and does not stop the spinner; maybe we should display a message to the user to draw attention to having not picked an audio file and stop the loading spinner

Fixed! There's now proper error handling. Thanks!

**Comment by ahelmberger** [comment]
> Works very nicely! clap
> 
> Just 2 tiny things I found while testing:
> 
> * the info icon overlaps a bit with the background in the dialog

I removed this tooltip/icon completely now as it was redundant (the same text is shown in the preview area initially)

> * when you select a wrong/non-audio format in the dialog from the CDN, it throws and does not stop the spinner; maybe we should display a message to the user to draw attention to having not picked an audio file and stop the loading spinner

Fixed! There's now proper error handling. Thanks!

**Review by adrianaluchian — APPROVED** [review]

**Pull request merged**